src/main/java/org/entando/kubernetes/validator/TenantValidator.java

```diff
@@ -0,0 +1,132 @@
+        final Map<String, Boolean> ids = new HashMap<>();
+
+        if (tenants != null && !tenants.isEmpty()) {
+            tenants.stream()
+                    .filter(cfg -> {
+                        boolean isDuplicated = ids.containsKey(cfg.getTenantCode());
+
+                        if (isDuplicated) {
+                            getErrorListForTenant(cfg.getTenantCode())
+                                    .add("tenant with FQDNs' " + cfg.getFqdns() + "' is using the same tenant id (" + cfg.getTenantCode() + ")");
+                        }
+                        return !isDuplicated;
+                    })
+                    .forEach(cfg -> ids.put(cfg.getTenantCode(), true));
+        }
+    }
+
+    private void validateFqdns(String fqdnsValueString, String tenantCode) {
+        if (StringUtils.isNotBlank(fqdnsValueString)) {
+            String[] fqdns = fqdnsValueString.split(",");
+            Arrays.asList(fqdns).forEach(fqdn -> {
+                if (!validateFQDN(fqdn)) {
+                    getErrorListForTenant(tenantCode).add("fqdns: invalid value detected '" + fqdn + "'");
+                }
+            });
+        }
+    }
+
+    private void validateFqdnsUniqueness(List<TenantConfigDTO> tenants) {
+        if (tenants != null && !tenants.isEmpty()) {
+            final Map<String, String> fqdns = new HashMap<>();
+
+            tenants.forEach(config -> {
+                final String tenantCode = config.getTenantCode();
+                final String fqdnsStr = config.getFqdns();
+                if (StringUtils.isNotBlank(fqdnsStr)) {
+                    String[] fqdnsarr = fqdnsStr.split(",");
+
+                    Arrays.asList(fqdnsarr).forEach(fqdn -> {
+                        if (!fqdns.containsKey(fqdn)) {
+                            fqdns.put(fqdn, tenantCode);
+                        } else {
+                            getErrorListForTenant(tenantCode).add("fqdns: '" + fqdn + "' already used for tenant '" + fqdns.get(fqdn) + "'");
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    private List<String> getErrorListForTenant(String id) {
+        if (validationErrors == null) {
+            validationErrors = new HashMap<>();
+        }
+        if (!validationErrors.containsKey(id)) {
+            final List<String> tenantValidationErrors = new ArrayList<>();
+
+            validationErrors.put(id, tenantValidationErrors);
+            return tenantValidationErrors;
+        }
+        return validationErrors.get(id);
+    }
+
+    public Optional<Map<String, List<String>>> getValidationErrorMap() {
+        if (validationErrors != null
+                && !validationErrors.isEmpty()) {
+            validationErrors.keySet().forEach(k -> {
+                log.error("Tenant '{}'", k);
+                validationErrors.get(k).forEach(e -> log.error("\t{}", e));
```

Comment:
you will not be sure the log will be in consecutive lines, you should put all in a unique log